src/test/shared/codelens/localLambdaRunner.test.ts

```diff
@@ -378,4 +385,51 @@ describe('localLambdaRunner', async () => {
             })
         }
     })
+
+    describe('makeBuildDir', () => {
+        it ('creates a temp directory', async () => {
+            const dir = await localLambdaRunner.makeBuildDir()
+            assert.strictEqual(fsUtils.fileExists(dir), true)
+        })
+    })
+
+    describe('executeSamBuild', () => {
+        const failedChildProcess: ChildProcessResult = {
+            exitCode: 1,
+            error: new Error('you are already dead'),
+            stdout: 'friendly failure message',
+            stderr: 'big ugly failure message'
+        }
+
+        const successfulChildProcess: ChildProcessResult = {
+            exitCode: 0,
+            error: undefined,
+            stdout: 'everything sunny all the time always',
+            stderr: 'nothing to report'
+        }
+
+        const generateSamBuildParams = (isSuccessfulBuild: boolean) => {
+            return {
+                baseBuildDir: tempDir,
+                codeDir: tempDir,
+                inputTemplatePath: tempDir,
+                channelLogger: new FakeChannelLogger(),
+                // not needed for testing
+                manifestPath: undefined,
+                samProcessInvoker: {
```

Comment:
What you do here with `samProcessInvoker` looks like a great set of tests to add to `samCliBuild.test.ts` as well.

Reply:
Will probably save that for later